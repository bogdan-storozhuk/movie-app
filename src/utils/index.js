const getFormatedGenresDescription = (genres) => {
  if (genres.length === 2) {
    return `${genres[0].name} & ${genres[1].name}`;
  } else if (genres.length >= 3) {
    return genres.reduceRight((accumulator, currentValue, index) => {
      let newValue = accumulator + currentValue.name;
      if (index === 0) {
        return newValue;
      }
      return `${newValue}, `;
    }, "");
  } else {
    return genres[0].name;
  }
};

const getGenreList = (withAll) => {
  const genreList = [
    { id: 1, name: "Drama" },
    { id: 2, name: "Romance" },
    { id: 3, name: "Comedy" },
    { id: 4, name: "Family" },
    { id: 5, name: "Adventure" },
    { id: 6, name: "Animation" },
    { id: 7, name: "Science Fiction" },
    { id: 8, name: "Fantasy" },
    { id: 9, name: "Action" },
    { id: 10, name: "Thriller" },
    { id: 11, name: "Mystery" },
  ];
  return withAll ? [{ id: 0, name: "All" }, ...genreList] : genreList;
};

const mapMovieJson = (movieData) => {
  return movieData.map((movie) => {
    const { vote_average, vote_count, release_date, poster_path, ...rest } =
      movie;
    return {
      ...rest,
      voteAverage: vote_average,
      voteCount: vote_count,
      releaseDate: release_date,
      posterPath: poster_path,
    };
  });
};

const mapMovieJsonToBackEndFormat = (value) => {
  if (value === "voteAverage") {
    return "vote_average";
  } else if (value === "voteCount") {
    return "vote_count";
  } else if (value === "releaseDate") {
    return "release_date";
  } else{
    return value
  }
};

const formatMoviesArray = (movies) => {
  return movies.map((movie) => {
    const { genres, ...rest } = movie;
    const updatedGenres = genres.map((genre) => ({ name: genre }));
    return { ...rest, genres: updatedGenres };
  });
};

export {
  getFormatedGenresDescription,
  getGenreList,
  mapMovieJson,
  formatMoviesArray,
  mapMovieJsonToBackEndFormat
};
