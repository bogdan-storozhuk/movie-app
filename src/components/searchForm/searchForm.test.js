import React from "react";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import SearchForm from "./";
import { renderWithRouter } from "../../utils";

describe("SearchForm", () => {
  const searchExample = "search example";
  it("should match snapshot with empty search value", () => {
    const { asFragment } = renderWithRouter(<SearchForm search="" />);

    expect(asFragment(<SearchForm />)).toMatchSnapshot();
  });

  it("should match snapshot with not empty search value", () => {
    const { asFragment } = renderWithRouter(
      <SearchForm search={searchExample} />
    );

    expect(asFragment(<SearchForm />)).toMatchSnapshot();
  });

  it("should correctly handle handleChange event", () => {
    renderWithRouter(<SearchForm search="" />);
    const input = screen.queryByPlaceholderText("What do you want to watch?");
    expect(input).toBeInTheDocument();
    userEvent.type(input, searchExample);
    expect(input.value).toBe(searchExample);
  });
});
