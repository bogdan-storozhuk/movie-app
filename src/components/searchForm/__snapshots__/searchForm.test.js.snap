// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match snapshot with empty search value 1`] = `
<DocumentFragment>
  <div
    class="Header-BottomPanel"
  >
    <h1
      class="Title"
    >
      FIND YOUR MOVIE
    </h1>
    <div
      class="search"
    >
      <input
        class="searchTerm"
        placeholder="What do you want to watch?"
        type="text"
        value=""
      />
      <a
        class="searchButton"
        href="/search/"
      >
        Search
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SearchForm should match snapshot with not empty search value 1`] = `
<DocumentFragment>
  <div
    class="Header-BottomPanel"
  >
    <h1
      class="Title"
    >
      FIND YOUR MOVIE
    </h1>
    <div
      class="search"
    >
      <input
        class="searchTerm"
        placeholder="What do you want to watch?"
        type="text"
        value="search example"
      />
      <a
        class="searchButton"
        href="/search/search example"
      >
        Search
      </a>
    </div>
  </div>
</DocumentFragment>
`;
